File: ruby/word-count/word_count.rb
require 'pry'

class Phrase
  attr_reader :str

  def initialize(str)
    @str = str
  end

  def word_count
    words = @str.split(" ")
    counts = Hash.new
    words.each do |word|
      # binding.pry
      counts[word] ||= 1
      # counts[word] += 1
    end
    counts
  end

end

File: java/word-count/src/main/java/WordCount.java
import java.util.HashMap;
import java.util.Map;
import java.util.Arrays;
import java.util.ArrayList;
import java.util.List;

public class WordCount {

  public Map<String, Integer> phrase(String phraseWords) {
    Map<String, Integer> wordCount = new HashMap<String, Integer>();

    String sanitized = phraseWords.replaceAll("[^a-zA-Z0-9\\s]", "").replaceAll("\\s+", " ").toLowerCase();
    String[] words = sanitized.split(" ");
    int i = 0;
    while (i < words.length) {
      if (wordCount.containsKey(words[i])) {
          int count = wordCount.get(words[i]);
          count++;
          wordCount.put(words[i], count);
      } else {
          wordCount.put(words[i], 1);
      }
      i++;
    }

    return wordCount;
  }

}
